srml/staking/README.adoc

```diff
@@ -0,0 +1,116 @@
+```
+
+Finally, account 4 can withdraw from any of the above roles via
+
+```rust
+Staking::chill(Origin::signed(4));
+```
+
+
+## Implementation Details
+
+### Slot Stake 
+
+The term `slot_stake` will be used throughout this section. It refers to a value calculated at the end of each era, containing the _minimum value at stake among all validators._
+
+### Reward Calculation 
+
+ - Rewards are recorded **per-session** and paid **per-era**. The value of reward for each session is calculated at the end of the session based on the timeliness of the session, then accumulated to be paid later. The value of the new _per-session-reward_ is calculated at the end of each era by multiplying `slot_stake` and a configuration storage named `SessionReward`. 
+ - Once a new era is triggered, rewards are paid to the validators and the associated nominators. 
+   - The validator can declare an amount that does not get shared with the nominators at each reward payout through their `ValidatorPrefs`. This value gets deducted from the total reward that can be paid. The remaining portion is split among the validator and all of the nominators who had a vote for this validator, proportional to their staked value. 
+  - All entities who receive a reward have the option to choose their reward destination, through the `Payee` storage, to be one of the following: 
+    - Controller account.
+    - Stash account, not increasing the staked value.
+    - Stash account, also increasing the staked value.
+
+### Slashing details 
+
+- A validator can be _reported_ to be offline at any point via `on_offline_validator` public function. 
+- Each validator declares how many times they can be _reported_ before it actually gets slashed via the `unstake_threshold` in `ValidatorPrefs`. On top of this, the module also introduces a `OfflineSlashGrace`, which applies to all validators and prevents them from getting immediately slashed.
+- Similar to the reward value, the slash value is updated at the end of each era by multiplying `slot_stake` and a configuration storage item, `OfflineSlash`.
+- Once a validator has been reported a sufficient amount of times, the actual value that gets deducted from that validator, and every single nominator that voted for it calculated by multiplying the result of the above point by `2.pow(unstake_threshold)`.
+  - If the previous overflow, then `slot_stake` is used.
+  - If the previous is more than what the validator/nominator has in stake, all of their stake is slashed (`.max(total_stake)` in other words).
+
+### Election algorithm details.
+
+Current election algorithm is implemented based on Phragmen. The reference implementation can be found [here](https://github.com/w3f/consensus/tree/master/NPoS).
+
+## Extensibility
+
+// Details that the user can modify or customize to make their own 
+
+## Dependencies
+
+### GenesisConfig
+
+See `chain_spec.rs` for a list of attributed that can be provided.
```

Comment:
Instead of linking to `chain_spec`, the correct would be to add a link to actual `GenesisConfig` struct: https://crates.parity.io/srml_staking/struct.GenesisConfig.html